packages/batch-poster-monitor/index.ts

```diff
@@ -161,21 +162,57 @@ const showAlert = (childChainInformation: ChainInfo, reasons: string[]) => {
   )
 }
 
+type EventLogs = Log<
+  bigint,
+  number,
+  false,
+  AbiEvent,
+  undefined,
+  [AbiEvent],
+  string
+>[]
+
+const getBatchPosterFromEventLogs = async (
+  eventLogs: EventLogs,
+  parentChainClient: PublicClient
+) => {
+  // get the batch-poster for the first event log
+  const batchPostingTransactionHash = eventLogs[0].transactionHash
+  const tx = await parentChainClient.getTransaction({
+    hash: batchPostingTransactionHash,
+  })
+  return tx.from
+}
+
 const getBatchPosterLowBalanceAlertMessage = async (
   parentChainClient: PublicClient,
-  childChainInformation: ChainInfo
+  childChainInformation: ChainInfo,
+  sequencerInboxLogs: EventLogs
 ) => {
-  //@ts-ignore - PublicClient that we pass vs PublicClient that orbit-sdk expects is not matching
-  const { batchPosters } = await getBatchPosters(parentChainClient, {
-    rollup: childChainInformation.ethBridge.rollup as `0x${string}`,
-    sequencerInbox: childChainInformation.ethBridge
-      .sequencerInbox as `0x${string}`,
-  })
-  if (!batchPosters || batchPosters.length === 0) {
-    return `Batch poster information not found`
-  }
+  let batchPoster: `0x${string}` | null = null
+
+  // try fetching batch poster address from orbit-sdk
+  try {
+    //@ts-ignore - PublicClient that we pass vs PublicClient that orbit-sdk expects is not matching
+    const { batchPosters } = await getBatchPosters(parentChainClient, {
+      rollup: childChainInformation.ethBridge.rollup as `0x${string}`,
+      sequencerInbox: childChainInformation.ethBridge
+        .sequencerInbox as `0x${string}`,
+    })
 
-  const batchPoster = batchPosters[0]
+    batchPoster = batchPosters[0] // get the first batch poster
+  } catch {
+    // else try fetching the batch poster from the event logs
+    try {
+      batchPoster = await getBatchPosterFromEventLogs(
+        sequencerInboxLogs,
+        parentChainClient
+      )
+    } catch {
+      // batchPoster not found by any means
+      return `Batch poster information not found`
```

Comment:
` is not needed here, we can use regular quote here

Reply:
Updated